### PR TITLE
add secret provider class support

### PR DESCRIPTION
**Pull request by ofir76**

### CODE REVIEW

File: templates/deployment.yaml [hunk 1]
```diff
@@ -38,9 +38,16 @@ spec:
...
               readOnly: true
+            {{- if .Values.secretProviderClass }}
+            - name: secrets-store-inline
+              readOnly: true
+              mountPath: /mnt/secrets-store
+            {{- end }}
           env:
             - name: RESOURCES_CONFIG_PATH
               value: "/etc/apono/resources_config.json"
+            - name: APONO_STORAGE_ENV_DIR
```
Comment (DmitriyO) [inline templates/deployment.yaml:49]: -> APONO_SECRETS_DIR

File: templates/deployment.yaml [hunk 2]
```diff
@@ -62,6 +69,14 @@ spec:
             items:
               - key: config
                 path: "resources_config.json"
+        {{- if .Values.secretProviderClass }}
+        - name: secrets-store-inline
+          csi:
+            driver: secrets-store.csi.k8s.io
+            readOnly: true
+            volumeAttributes:
+              secretProviderClass: {{ .Values.secretProviderClass }}
```
Comment (DmitriyO) [inline templates/deployment.yaml:78]: what value is going into here?

File: values.yaml [hunk 3]
```diff
@@ -28,6 +28,8 @@ podAnnotations: {}
 podSecurityContext: {}
   # fsGroup: 2000
 
+secretProviderClass: ""
```
Comment (DmitriyO) [inline values.yaml:31]: no default value? what are the possible values?

File: templates/deployment.yaml (hunk 2 above)
Comment (ofir76) [reply templates/deployment.yaml:78]: the secret provider class name

File: values.yaml (hunk 3 above)
Comment (ofir76) [reply values.yaml:31]: name of the secret provider class...

Comment (ofir76) [reply values.yaml:31]: you can read in the apono-agent readme tell me if its make sense

**Pull request merged**